hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/AWSCredentialProviderList.java

```diff
@@ -58,10 +58,14 @@
  *   rethrown; exceptions other than 'no credentials' have priority.</li>
  *   <li>Special handling of {@link AnonymousAWSCredentials}.</li>
  * </ol>
+ *
+ * @deprecated This class will be replaced by one that implements AWS SDK V2's AwsCredentialProvider
+ * as part of upgrading S3A to SDK V2. See HADOOP-18073.
  */
 @InterfaceAudience.Private
 @InterfaceStability.Evolving
-public class AWSCredentialProviderList implements AWSCredentialsProvider,
+@Deprecated
```

Comment:
i am proposing the class does implement the v2 api. so no need to deprecate or break the delegation token binding

Reply:
have removed this deprecation. Since this means DT bindings will not break, and custom cred providers there can continue to use V1 cred providers, do we want to [remove the warning on use of DT's](https://github.com/apache/hadoop/pull/4698/files#diff-2dcd9ad45ff8e3a6e5ed6cad454f2e8a5beb42e30291a785f425da70f40c3fb3R74)? I guess we still want to encourage people to update these credential providers so might be worth leaving in.